### PR TITLE
Updating mysqli: mysqli_stmt_num_rows

### PR DESCRIPTION
**Pull request by kamil-tekiela**
Improved the page to cause less confusion. I am unsure about the page title. I left it almost unchanged but maybe it should be changed too?

### CODE REVIEW

File: reference/mysqli/mysqli_stmt/num-rows.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <refname>mysqli_stmt::$num_rows</refname>
   <refname>mysqli_stmt::num_rows</refname>
   <refname>mysqli_stmt_num_rows</refname>
-  <refpurpose>Return the number of rows in statements result set</refpurpose>
+  <refpurpose>Return the number of rows in statement&apos;s result set</refpurpose>
```
Comment (cmb69) [inline reference/mysqli/mysqli_stmt/num-rows.xml:8]: I think we can drop "statement's" altogether.

File: reference/mysqli/mysqli_stmt/num-rows.xml [hunk 2]
```diff
@@ -22,15 +22,13 @@
    <methodparam><type>mysqli_stmt</type><parameter>statement</parameter></methodparam>
   </methodsynopsis>
   <para>
-   Returns the number of rows in the result set.
-   The use of <function>mysqli_stmt_num_rows</function>
-   depends on whether or not you used
+   Returns the number of buffered rows.
+   The use of this function depends on whether you used
```
Comment (cmb69) [inline reference/mysqli/mysqli_stmt/num-rows.xml:26]: Please no "you" on manual pages; passive voice seems suitable here.

File: reference/mysqli/mysqli_stmt/num-rows.xml [hunk 3]
```diff
@@ -22,15 +22,13 @@
...
    <function>mysqli_stmt_store_result</function> to buffer the entire result
    set in the statement handle.
   </para>
   <para>
-   If you use <function>mysqli_stmt_store_result</function>,
-   <function>mysqli_stmt_num_rows</function> may be called immediately.
+   This function returns 0 unless all rows have been fetched from the server.
```
Comment (cmb69) [inline reference/mysqli/mysqli_stmt/num-rows.xml:31]: Nit:
```suggestion
   This function returns <literal>0</literal> unless all rows have been fetched from the server.
```

File: reference/mysqli/mysqli_stmt/num-rows.xml [hunk 4]
```diff
@@ -46,7 +44,8 @@
  <refsect1 role="returnvalues">
   &reftitle.returnvalues;
   <para>
-   An integer representing the number of rows in result set.
+   An integer representing the number of buffered rows. Returns 0 in unbuffered
```
Comment (cmb69) [inline reference/mysqli/mysqli_stmt/num-rows.xml:47]:
```suggestion
   An <type>int</type> representing the number of buffered rows. Returns <literal>0</literal> in unbuffered
```

File: reference/mysqli/mysqli_stmt/num-rows.xml [hunk 5]
```diff
@@ -46,7 +44,8 @@
...
+   mode if all rows have not yet been fetched from the server.
```
Comment (cmb69) [inline reference/mysqli/mysqli_stmt/num-rows.xml:48]: This sentence reads strange to me. Maybe better: "if not all rows have yet been fetched from the server." Unsure.

**Review by cmb69 — COMMENTED** [review]
Thanks you for the PR. Looks generally good to me, but a few improvements might be appropriate.

File: reference/mysqli/mysqli_stmt/num-rows.xml (hunk 1 above)
Comment (kamil-tekiela) [reply reference/mysqli/mysqli_stmt/num-rows.xml:8]: I am not sure about this. It seems important to me to distinguish the two `num_rows` in `mysqli_stmt` and `mysqli_result`. I have seen people confuse them a lot. I'll think about this sentence and maybe I will rewrite it.

File: reference/mysqli/mysqli_stmt/num-rows.xml (hunk 2 above)
Comment (kamil-tekiela) [reply reference/mysqli/mysqli_stmt/num-rows.xml:26]: I missed that. I'll fix it.

**Comment by kamil-tekiela** [comment]
I redid the PR. I used simpler English than what was used before. I strayed away from the original https://dev.mysql.com/doc/c-api/8.0/en/mysql-stmt-num-rows.html but I believe this explains this function much better than MySQL's original description.

**Review by cmb69 — APPROVED** [review]
Great, thank you!

**Pull request merged**